src/suites/cts/validation/createView.spec.ts

```diff
@@ -0,0 +1,268 @@
+
+  await t.expectValidationError(() => {
+    texture.createView(descriptor);
+  }, !success);
+}).params([
+  { success: true },
+  { format: 'rgba8unorm', success: true },
+  { format: 'r8unorm', success: false },
+  { dimension: '2d-array', success: true },
+  { dimension: '2d', success: true },
+  { arrayLayerCount: 0, success: true },
+  { arrayLayerCount: 1, success: true },
+  { arrayLayerCount: 2, success: false },
+  { mipLevelCount: MIP_LEVEL_COUNT, success: true },
+  { mipLevelCount: 1, success: true },
+]);
+
+g.test('creating cube map texture view', async t => {
+  const { dimension = '2d-array', arrayLayerCount, success } = t.params;
+
+  const texture = t.createTexture({ arrayLayerCount: 16 });
+
+  const descriptor = t.getDescriptor({
+    dimension,
+    arrayLayerCount,
+  });
+
+  await t.expectValidationError(() => {
+    texture.createView(descriptor);
+  }, !success);
+}).params([
+  { dimension: 'cube', arrayLayerCount: 6, success: true }, // it is OK to create a cube map texture view with arrayLayerCount == 6
+  // it is an error to create a cube map texture view with arrayLayerCount != 6
+  { dimension: 'cube', arrayLayerCount: 3, success: false },
+  { dimension: 'cube', arrayLayerCount: 7, success: false },
+  { dimension: 'cube', arrayLayerCount: 12, success: false },
+  { dimension: 'cube-array', arrayLayerCount: 12, success: true }, // it is OK to create a cube map array texture view with arrayLayerCount % 6 == 0
+  // it is an error to create a cube map array texture view with arrayLayerCount % 6 != 0
+  { dimension: 'cube-array', arrayLayerCount: 11, success: false },
+  { dimension: 'cube-array', arrayLayerCount: 13, success: false },
+]);
+
+g.test('creating cube map texture view with a non square texture', async t => {
+  const { dimension, arrayLayerCount } = t.params;
+
+  const nonSquareTexture = t.createTexture({
+    arrayLayerCount: 18,
+    width: 32,
+    height: 16,
+    mipLevelCount: 5,
+  });
+
+  const descriptor = t.getDescriptor({
+    dimension,
+    arrayLayerCount,
+  });
+
+  await t.expectValidationError(() => {
+    nonSquareTexture.createView(descriptor);
+  });
+}).params([
+  { dimension: 'cube', arrayLayerCount: 6 }, // it is an error to create a cube map texture view with width != height.
+  { dimension: 'cube-array', arrayLayerCount: 12 }, // it is an error to create a cube map array texture view with width != height.
+]);
+
+// TODO: add more tests when rules are fully implemented.
+g.test('test the format compatibility rules when creating a texture view', async t => {
+  const texture = t.createTexture({ arrayLayerCount: 1 });
+
+  const descriptor = t.getDescriptor({
+    format: 'depth24plus-stencil8',
+  });
+
+  // it is invalid to create a view in depth-stencil format on a RGBA texture
+  await t.expectValidationError(() => {
+    texture.createView(descriptor);
+  });
+});
+
+g.test('it is invalid to use a texture view created from a destroyed texture', async t => {
```

Comment:
comment needs to be updated?

Reply:
@kainino0x said it was valid to create a texture view from a destroyed texture, just invalid to use later. See https://github.com/gpuweb/cts/pull/41#discussion_r329845848

Is `it is invalid to use a texture view created from a destroyed texture` OK then?